[Bug]: Local build fails during java test phase on MacOS/aarch64

### Bug Description

On MacOS on Apple M2 (aarch64) with Strimzi versions newer than 0.33.x, builds fail during java_build stage due to multiple tests failing with timeout errors like the following:
```
java.lang.AssertionError: 
Expected: an instance of io.fabric8.kubernetes.client.KubernetesClientException
     but: <io.strimzi.operator.common.operator.resource.TimeoutException: Exceeded timeout of 10000ms while waiting for KafkaBridge resource my-test-resource-896958757 in namespace bridge-crd-it-namespace to be ready> is a io.strimzi.operator.common.operator.resource.TimeoutException
```
It seems that according to Minikube, these resources do eventually become ready, though not necessarily always within the timeout period.

Building Strimzi version 0.33.2 does not exhibit this behaviour, and the tests and build succeed every time.

Notably, increasing memory and/or CPU allocation for Minikube from default settings does not seem to have any effect on this issue (even quadrupling the default values does not have any noticeable effect), nor does allocating more nodes to the Minikube cluster.

### Steps to reproduce

1. Pull repository
2. Run `mvn clean install -DskipTests`
3. Run `make java_build` or `make all`
4. Build will fail at component operator-common with multiple test failures due to timeouts waiting for resources on Minikube.

### Expected behavior

Build should not fail on newer versions of Strimzi.

### Strimzi version

0.34.x, 0.35.x

### Kubernetes version

Kubernetes 1.26.3

### Installation method

_N/A_

### Infrastructure

Containerd 1.6.20, Minikube 1.30.1, Podman 4.5.0, MacOS Ventura 13.3.1 (a) on Apple M2

### Configuration files and logs

Console output from working build of Strimzi 0.33.2: [20230518T143139_strimzi_0.33.2_build.log](https://github.com/strimzi/strimzi-kafka-operator/files/11504785/20230518T143139_strimzi_0.33.2_build.log)
Console output from broken build of Strimzi 0.35.0: [20230518T174413_strimzi_0.35.0_build.log](https://github.com/strimzi/strimzi-kafka-operator/files/11504787/20230518T174413_strimzi_0.35.0_build.log)


### Additional context

Note that running Minikube with the default Docker container engine and Docker driver on aarch64 causes the image build stage to fail due to seperate issues with Docker on this architecture. Running Minikube with Podman driver and Containerd engine works for this purpose, however, so long as you also use socket_vmnet. (I was planning to submit a PR to update the Dev Guide with this info once I figured out what was going on with the java build stage...)

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java
@@ -306,7 +306,8 @@ public class Main {
             Future.all(futures).onComplete(res -> {
                 if (res.succeeded())    {
                     returnPromise.complete();
-                } else  {
+                } else {
+                    LOGGER.error("Failed to create Cluster Roles.", res.cause());
                     returnPromise.fail("Failed to create Cluster Roles.");
                 }
             });

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/StrimziPodSetControllerIT.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/StrimziPodSetControllerIT.java
@@ -319,11 +319,16 @@ public class StrimziPodSetControllerIT {
             // Delete the PodSet
             podSetOp().inNamespace(NAMESPACE).withName(podSetName).delete();
 
-            // Check that pod is deleted after pod set deletion
+            /*
+             * Check that pod is deleted after pod set deletion
+             *
+             * The CR finalizer removal does not complete in a consistent amount of time,
+             * therefore the timeout for waiting on the deletion is increased to 1 minute.
+             */
             TestUtils.waitFor(
                     "Wait for Pod to be deleted",
                     100,
-                    10_000,
+                    60_000,
                     () -> client.pods().inNamespace(NAMESPACE).withName(podName).get() == null,
                     () -> context.failNow("Test timed out waiting for pod deletion!"));
 
@@ -655,11 +660,16 @@ public class StrimziPodSetControllerIT {
             // Delete the PodSet in non-cascading way
             podSetOp().inNamespace(NAMESPACE).withName(podSetName).withPropagationPolicy(DeletionPropagation.ORPHAN).delete();
 
-            // Check that the PodSet is deleted
+            /*
+             * Check that the PodSet is deleted
+             *
+             * The CR finalizer removal does not complete in a consistent amount of time,
+             * therefore the timeout for waiting on the deletion is increased to 1 minute.
+             */
             TestUtils.waitFor(
                     "Wait for StrimziPodSet deletion",
                     100,
-                    10_000,
+                    60_000,
                     () -> {
                         StrimziPodSet podSet = podSetOp().inNamespace(NAMESPACE).withName(podSetName).get();
                         return podSet == null;

--- a/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/CrdOperator.java
+++ b/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/CrdOperator.java
@@ -74,11 +74,11 @@ public class CrdOperator<C extends KubernetesClient,
             "deleted",
             1_000,
             deleteTimeoutMs(),
-            () -> resourceOp.get() != null);
+            () -> resourceOp.get() == null);
 
         Future<Void> deleteFuture = resourceSupport.deleteAsync(resourceOp.withPropagationPolicy(cascading ? DeletionPropagation.FOREGROUND : DeletionPropagation.ORPHAN).withGracePeriod(-1L));
 
-        return Future.join(watchForDeleteFuture, deleteFuture).map(ReconcileResult.deleted());
+        return Future.all(watchForDeleteFuture, deleteFuture).map(ReconcileResult.deleted());
     }
 
     /**

--- a/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/concurrent/CrdOperator.java
+++ b/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/concurrent/CrdOperator.java
@@ -78,7 +78,7 @@ public class CrdOperator<C extends KubernetesClient,
             "deleted",
             1_000,
             deleteTimeoutMs(),
-            () -> resourceOp.get() != null)
+            () -> resourceOp.get() == null)
             .toCompletableFuture();
 
         CompletableFuture<Void> deleteFuture = resourceSupport.deleteAsync(resourceOp.withPropagationPolicy(cascading ? DeletionPropagation.FOREGROUND : DeletionPropagation.ORPHAN).withGracePeriod(-1L))

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java
@@ -18,6 +18,7 @@ import io.vertx.core.Promise;
 import io.vertx.core.Vertx;
 import io.vertx.core.WorkerExecutor;
 import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.Timeout;
 import io.vertx.junit5.VertxExtension;
 import io.vertx.junit5.VertxTestContext;
 import org.apache.logging.log4j.LogManager;
@@ -29,6 +30,7 @@ import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.util.Random;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
@@ -108,6 +110,8 @@ public abstract class AbstractCustomResourceOperatorIT<C extends KubernetesClien
             LOGGER.warn("Deleting namespace {} after tests run", namespace);
             kubeClient().deleteNamespace(namespace);
             cmdKubeClient().waitForResourceDeletion("Namespace", namespace);
+            LOGGER.info("Deleting CRD");
+            KubeClusterResource.getInstance().deleteCustomResources();
         }
     }
 
@@ -142,11 +146,15 @@ public abstract class AbstractCustomResourceOperatorIT<C extends KubernetesClien
     }
 
     /**
-     * Tests what happens when the resource is deleted while updating the status
+     * Tests what happens when the resource is deleted while updating the status.
      *
-     * @param context   Test context
+     * The CR removal does not consistently complete within the default timeout.
+     * This requires increasing the timeout for completion to 1 minute.
+     *
+     * @param context Test context
      */
     @Test
+    @Timeout(value = 1, timeUnit = TimeUnit.MINUTES)
     public void testUpdateStatusAfterResourceDeletedThrowsKubernetesClientException(VertxTestContext context) {
         String resourceName = getResourceName(RESOURCE_NAME);
         Checkpoint async = context.checkpoint();
@@ -167,10 +175,7 @@ public abstract class AbstractCustomResourceOperatorIT<C extends KubernetesClien
                     return op.deleteAsync(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, false);
                 })
                 .onComplete(context.succeeding(i -> { }))
-                .compose(i -> {
-                    LOGGER.info("Wait for confirmed deletion");
-                    return op.waitFor(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, 100L, 10_000L, (n, ns) -> operator().get(namespace, resourceName) == null);
-                })
+
                 .compose(i -> {
                     LOGGER.info("Updating resource with new status - should fail");
                     return op.updateStatusAsync(Reconciliation.DUMMY_RECONCILIATION, newStatus.get());

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNamespacedResourceOperatorTest.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNamespacedResourceOperatorTest.java
@@ -296,13 +296,17 @@ public abstract class AbstractNamespacedResourceOperatorTest<C extends Kubernete
     @Test
     public void testReconcileDeleteWhenResourceExistsStillDeletes(VertxTestContext context) {
         Deletable mockDeletable = mock(Deletable.class);
-        when(mockDeletable.delete()).thenReturn(List.of());
+        AtomicBoolean resourceDeleted = new AtomicBoolean(false);
+        when(mockDeletable.delete()).thenAnswer(args -> {
+            resourceDeleted.set(true);
+            return List.of();
+        });
         GracePeriodConfigurable mockDeletableGrace = mock(GracePeriodConfigurable.class);
         when(mockDeletableGrace.delete()).thenReturn(List.of());
 
         T resource = resource();
         Resource mockResource = mock(resourceType());
-        when(mockResource.get()).thenReturn(resource);
+        when(mockResource.get()).thenAnswer(args -> resourceDeleted.get() ? null : resource);
         when(mockResource.withPropagationPolicy(eq(DeletionPropagation.FOREGROUND))).thenReturn(mockDeletableGrace);
         when(mockDeletableGrace.withGracePeriod(anyLong())).thenReturn(mockDeletable);
         AtomicBoolean watchClosed = new AtomicBoolean(false);
@@ -336,13 +340,17 @@ public abstract class AbstractNamespacedResourceOperatorTest<C extends Kubernete
     @Test
     public void testReconcileDeletionSuccessfullyDeletes(VertxTestContext context) {
         Deletable mockDeletable = mock(Deletable.class);
-        when(mockDeletable.delete()).thenReturn(List.of());
+        AtomicBoolean resourceDeleted = new AtomicBoolean(false);
+        when(mockDeletable.delete()).thenAnswer(args -> {
+            resourceDeleted.set(true);
+            return List.of();
+        });
         GracePeriodConfigurable mockDeletableGrace = mock(GracePeriodConfigurable.class);
         when(mockDeletableGrace.delete()).thenReturn(List.of());
 
         T resource = resource();
         Resource mockResource = mock(resourceType());
-        when(mockResource.get()).thenReturn(resource);
+        when(mockResource.get()).thenAnswer(args -> resourceDeleted.get() ? null : resource);
         when(mockResource.withPropagationPolicy(eq(DeletionPropagation.FOREGROUND))).thenReturn(mockDeletableGrace);
         when(mockDeletableGrace.withGracePeriod(anyLong())).thenReturn(mockDeletable);
         AtomicBoolean watchClosed = new AtomicBoolean(false);
@@ -417,13 +425,17 @@ public abstract class AbstractNamespacedResourceOperatorTest<C extends Kubernete
     @SuppressWarnings("unchecked")
     public void testReconcileDeleteDoesNotThrowWhenDeletionReturnsFalse(VertxTestContext context) {
         Deletable mockDeletable = mock(Deletable.class);
-        when(mockDeletable.delete()).thenReturn(List.of());
+        AtomicBoolean resourceDeleted = new AtomicBoolean(false);
+        when(mockDeletable.delete()).thenAnswer(args -> {
+            resourceDeleted.set(true);
+            return List.of();
+        });
         GracePeriodConfigurable mockDeletableGrace = mock(GracePeriodConfigurable.class);
         when(mockDeletableGrace.delete()).thenReturn(List.of());
 
         T resource = resource();
         Resource mockResource = mock(resourceType());
-        when(mockResource.get()).thenReturn(resource);
+        when(mockResource.get()).thenAnswer(args -> resourceDeleted.get() ? null : resource);
         when(mockResource.withPropagationPolicy(eq(DeletionPropagation.FOREGROUND))).thenReturn(mockDeletableGrace);
         when(mockDeletableGrace.withGracePeriod(anyLong())).thenReturn(mockDeletable);
         AtomicBoolean watchClosed = new AtomicBoolean(false);
@@ -518,7 +530,11 @@ public abstract class AbstractNamespacedResourceOperatorTest<C extends Kubernete
 
         // For resource1 we need to mock the async deletion process as well
         Deletable mockDeletable1 = mock(Deletable.class);
-        when(mockDeletable1.delete()).thenReturn(List.of());
+        AtomicBoolean resource1Deleted = new AtomicBoolean(false);
+        when(mockDeletable1.delete()).thenAnswer(args -> {
+            resource1Deleted.set(true);
+            return List.of();
+        });
         GracePeriodConfigurable mockDeletableGrace1 = mock(GracePeriodConfigurable.class);
         when(mockDeletableGrace1.withGracePeriod(anyLong())).thenReturn(mockDeletable1);
         Resource mockResource1 = mock(resourceType());
@@ -527,7 +543,7 @@ public abstract class AbstractNamespacedResourceOperatorTest<C extends Kubernete
         when(mockResource1.get()).thenAnswer(invocation -> {
             // First get needs to return the resource to trigger deletion
             // Next gets return null since the resource was already deleted
-            if (watchCreated.get()) {
+            if (watchCreated.get() || resource1Deleted.get()) {
                 return null;
             } else {
                 return resource1;

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/StrimziPodSetCrdOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/StrimziPodSetCrdOperatorIT.java
@@ -19,6 +19,7 @@ import io.strimzi.operator.common.Reconciliation;
 import io.strimzi.test.TestUtils;
 import io.vertx.core.Promise;
 import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.Timeout;
 import io.vertx.junit5.VertxExtension;
 import io.vertx.junit5.VertxTestContext;
 import org.apache.logging.log4j.LogManager;
@@ -28,6 +29,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -190,11 +192,15 @@ public class StrimziPodSetCrdOperatorIT extends AbstractCustomResourceOperatorIT
     }
 
     /**
-     * Tests what happens when the resource is deleted while updating the status
+     * Tests what happens when the resource is deleted while updating the status.
      *
-     * @param context   Test context
+     * The CR removal does not consistently complete within the default timeout.
+     * This requires increasing the timeout for completion to 1 minute.
+     *
+     * @param context Test context
      */
     @Test
+    @Timeout(value = 1, timeUnit = TimeUnit.MINUTES)
     public void testUpdateStatusAfterResourceDeletedThrowsKubernetesClientException(VertxTestContext context) {
         String resourceName = getResourceName(RESOURCE_NAME);
         Checkpoint async = context.checkpoint();
@@ -215,10 +221,6 @@ public class StrimziPodSetCrdOperatorIT extends AbstractCustomResourceOperatorIT
                     return op.reconcile(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, null);
                 })
                 .onComplete(context.succeeding(i -> { }))
-                .compose(i -> {
-                    LOGGER.info("Wait for confirmed deletion");
-                    return op.waitFor(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, 100L, 10_000L, (n, ns) -> operator().get(namespace, resourceName) == null);
-                })
                 .compose(i -> {
                     LOGGER.info("Updating resource with new status - should fail");
                     return op.updateStatusAsync(Reconciliation.DUMMY_RECONCILIATION, newStatus.get());

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/concurrent/AbstractCustomResourceOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/concurrent/AbstractCustomResourceOperatorIT.java
@@ -24,6 +24,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 
 import java.util.Random;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
@@ -126,7 +127,10 @@ public abstract class AbstractCustomResourceOperatorIT<
     }
 
     /**
-     * Tests what happens when the resource is deleted while updating the status
+     * Tests what happens when the resource is deleted while updating the status.
+     *
+     * The CR removal does not consistently complete within the default timeout.
+     * This requires increasing the timeout for completion to 1 minute.
      */
     @Test
     public void testUpdateStatusAfterResourceDeletedThrowsKubernetesClientException() {
@@ -145,17 +149,15 @@ public abstract class AbstractCustomResourceOperatorIT<
                 return op.deleteAsync(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, false);
             })
             .thenCompose(i -> {
-                LOGGER.info("Wait for confirmed deletion");
-                return op.waitFor(Reconciliation.DUMMY_RECONCILIATION, namespace, resourceName, 100L, 10_000L, (n, ns) -> operator().get(namespace, resourceName) == null);
-            })
-            .thenCompose(i -> {
                 LOGGER.info("Updating resource with new status - should fail");
                 return op.updateStatusAsync(Reconciliation.DUMMY_RECONCILIATION, newStatus.get());
             })
             .<Void>handle((i, error) -> {
                 assertThat(Util.unwrap(error), instanceOf(KubernetesClientException.class));
                 return null;
-            }));
+            }),
+            1,
+            TimeUnit.MINUTES);
     }
 
     /**

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/concurrent/AbstractNamespacedResourceOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/concurrent/AbstractNamespacedResourceOperatorIT.java
@@ -123,7 +123,7 @@ public abstract class AbstractNamespacedResourceOperatorIT<
                     "resource deletion " + resourceName,
                     "deleted",
                     1000,
-                    30_000,
+                    60_000,
                     () -> op.get(namespace, resourceName) == null))
             .thenRun(() -> {
                 assertThat(op.get(namespace, resourceName), is(nullValue()));

